libs/application/templates/children-residence-change/src/fields/Overview/SignatureModal/index.tsx

```diff
@@ -38,22 +83,77 @@ const SignatureModal = ({
     >
-      <Box
-        className={style.modalContent}
-        boxShadow="large"
-        borderRadius="standard"
-        background="white"
-        paddingX={[3, 3, 5, 12]}
-        paddingY={[3, 3, 4, 10]}
-      >
-        <Text variant="h1" marginBottom={2}>
-          {formatMessage(signatureModal.general.title)}
+      <Box className={styles.modalContent}>
+        <Box className={styles.logoWrapper}>
+          <Logo id="modal" iconOnly={true} />
+        </Box>
+        <Text variant="h2" marginBottom={1} lineHeight="md">
+          {formatMessage(fileSignatureState.content.title)}
         </Text>
-        <ModalConditionalContent
-          fileSignatureState={fileSignatureState}
-          onClose={onClose}
-          controlCode={controlCode}
-        />
+        {isRequest ? (
+          <>
+            <SkeletonLoader height={20} space={1} />
+            <SkeletonLoader height={20} width="50%" space={1} />
+          </>
+        ) : (
+          fileSignatureState.content?.message && (
+            <Text>{formatMessage(fileSignatureState.content.message)}</Text>
+          )
+        )}
+        <Box
+          className={cn(styles.controlCodeContainer.general, {
+            [styles.controlCodeContainer.upload]: isUpload,
+            [styles.controlCodeContainer.error]: hasError,
+          })}
+          marginTop={4}
+          background={
+            statusVariantStyles[fileSignatureState.content.type].background
+          }
+        >
+          {!isRequest &&
+            (isUpload ? (
+              <>
+                <Text lineHeight="xl" variant="small" marginTop={2}>
+                  {formatMessage(signatureModal.security.numberLabel)}
+                </Text>
+                <Text lineHeight="sm" variant="h1">
+                  {controlCode}
+                </Text>
+              </>
+            ) : (
+              <Icon
+                className={styles.iconContainer}
+                color={
+                  statusVariantStyles[fileSignatureState.content.type].iconColor
+                }
+                icon={statusVariantStyles[fileSignatureState.content.type].icon}
+                size="large"
+                type="filled"
+              />
+            ))}
+        </Box>
+        <Box marginTop={6} display="flex" justifyContent="center">
+          {hasError ? (
+            <Button
+              onClick={onClose}
+              fluid={true}
+              variant="ghost"
+              colorScheme="destructive"
+            >
+              {formatMessage(signatureModal.general.closeButtonLabel)}
+            </Button>
+          ) : (
+            <div
+              className={cn(styles.loader.general, {
+                [styles.loader.noLoader]: isRequest,
+              })}
+            >
+              <div className={styles.loadingDot} />
+              <div className={styles.loadingDot} />
+              <div className={styles.loadingDot} />
+            </div>
+          )}
```

Comment:
We could rip some of this stuff out to smaller components, I went back and forth if it was a good idea, in the end I feel like it is not needed.  Do you have any thoughts about it?

Reply:
Yeah I was actually going to mention this to Embla. I feel like this loader should for example be a part of the island-ui library. But I wen back and forth also and decided that we didn't need to do that in this PR.